daemon/logger/adapter.go

```diff
@@ -93,21 +93,12 @@ func (a *pluginAdapterWithRead) ReadLogs(config ReadConfig) *LogWatcher {
 
 		dec := logdriver.NewLogEntryDecoder(stream)
 		for {
-			select {
-			case <-watcher.WatchClose():
-				return
-			default:
-			}
-
 			var buf logdriver.LogEntry
 			if err := dec.Decode(&buf); err != nil {
 				if err == io.EOF {
 					return
 				}
-				select {
-				case watcher.Err <- errors.Wrap(err, "error decoding log message"):
-				case <-watcher.WatchClose():
-				}
+				watcher.Err <- errors.Wrap(err, "error decoding log message")
```

Comment:
This should be done on a select or else this can block.

Reply:
I have to admit I don't understand it :(

Here's what I see:

1. Since this channel is made with `make chan(error, 1)` it is buffered, and the buffer size is 1. So it would block if we'll send more than 1 error before someone receives it. I do not see how can we send more than 1 error from here.
 
2. In the same function, about 20 lines above this, there's a blocking send to the same channel.

Reply:
If there's a buffer, this should be fine.
It is true that the expectation is if there is an error sent that it should return.
I have considered using this to send errors for data corruption and attempting a recovery (by e.g. scanning ahead to the next json object)... but that's not the current behavior.